### PR TITLE
[BEAM-6914] Reverting behavior of Native BQ sink in Python

### PR DESCRIPTION
**Pull request by pabloem**
r: @chamikaramj 
cc: @tvalentyn 

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PreCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
@chamikaramj this should leave the BQ IO module in a good state to behave as it was in 2.11.

File: sdks/python/apache_beam/io/gcp/bigquery_tools.py [hunk 1]
```diff
@@ -75,6 +75,15 @@ def default_encoder(obj):
       "Object of type '%s' is not JSON serializable" % type(obj).__name__)
 
 
+def get_hashable_destination(destination):
```
Comment (tvalentyn) [inline sdks/python/apache_beam/io/gcp/bigquery_tools.py]: https://github.com/apache/beam/pull/8076 adds a similar change and has a docstring for this function, consider copying the docstring from that PR.

File: sdks/python/apache_beam/io/gcp/bigquery_file_loads_test.py [hunk 2]
```diff
@@ -393,7 +393,8 @@ def test_multiple_destinations_transform(self):
                   if 'foundation' in d])]
 
     args = self.test_pipeline.get_full_options_as_args(
-        on_success_matcher=all_of(*pipeline_verifiers))
+        on_success_matcher=all_of(*pipeline_verifiers),
+        experiments='new_bq_sink')
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/bigquery_file_loads_test.py]: How about use_beam_bq_sink

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 3]
```diff
@@ -590,6 +591,32 @@ def run_Flatten(self, transform_node, options):
           PropertyNames.ENCODING: step.encoding,
           PropertyNames.OUTPUT_NAME: PropertyNames.OUT}])
 
+  def apply_WriteToBigQuery(self, transform, pcoll, options):
```
Comment (chamikaramj) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: Please add a TODO to delete this method once new BQ sink is the default (and prob. also create a JIRA)

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 4]
```diff
@@ -590,6 +591,32 @@ def run_Flatten(self, transform_node, options):
...
+    else:
+      from apache_beam.io.gcp.bigquery_tools import parse_table_schema_from_json
+      return pcoll  | 'WriteToBigQuery' >> beam.io.Write(
+          beam.io.BigQuerySink(
+              transform.table_reference.tableId,
+              transform.table_reference.datasetId,
+              transform.table_reference.projectId,
+              parse_table_schema_from_json(json.dumps(transform.schema)),
+              transform.create_disposition,
+              transform.write_disposition,
```
Comment (chamikaramj) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: How about validate and coder parameters ?
https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/gcp/bigquery.py#L397

Comment (pabloem) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: This method is almost an exact replica of the way the transform was inserted before. Coder and validate arguments were left to their defaults.

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py (hunk 3 above)
Comment (pabloem) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: Done.

File: sdks/python/apache_beam/io/gcp/bigquery_file_loads_test.py (hunk 2 above)
Comment (pabloem) [reply sdks/python/apache_beam/io/gcp/bigquery_file_loads_test.py]: Done.

File: sdks/python/apache_beam/io/gcp/bigquery_tools.py (hunk 1 above)
Comment (pabloem) [reply sdks/python/apache_beam/io/gcp/bigquery_tools.py]: Done.

**Comment by pabloem** [comment]
@chamikaramj Done. Thanks.

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PreCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PreCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PreCommit

**Comment by pabloem** [comment]
@chamikaramj tests are passing now.

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Pull request merged**